cmd/docker/main.go

```diff
@@ -130,16 +136,30 @@ func handleFlags() bool {
 	}
 	extraconfig.Decode(src, &vchConfig)
 
-	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
-		log.SetLevel(log.DebugLevel)
-	}
-
 	*cli.portLayerAddr = fmt.Sprintf("%s:%d", *cli.portLayerAddr, *cli.portLayerPort)
 	cli.proto = "tcp"
 
 	return true
 }
 
+func initLogging() error {
+	logcfg := viclog.NewLoggingConfig()
+	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
+		logcfg.Level = log.DebugLevel
+	}
+
+	if vchConfig.Diagnostics.SysLogConfig != nil {
+		logcfg.Syslog = &syslog.SyslogConfig{
+			Network:   vchConfig.Diagnostics.SysLogConfig.Network,
+			RAddr:     vchConfig.Diagnostics.SysLogConfig.RAddr,
+			Formatter: syslog.RFC3164,
+			Priority:  srslog.LOG_INFO | srslog.LOG_DAEMON,
+		}
+	}
+
+	return viclog.Init(logcfg)
```

Comment:
We are pushing it onto a queue; the hook `Fire` call is not async, which is why I just put the entry into a buffered channel for precisely the reason you point out.

The timeout is OK, we just drop logs if that happens; no blocking happens of the actual log call. I don't want to make this very complicated -- for now no connection, we drop stuff is enough.